### ISSUE TITLE
Upgrade the Jackson dependencies to >= 2.9.4

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#618]
Due to vulnerabilities in our current version of Jackson, we need to upgrade to 2.9.4.

Unfortunately, just bumping the version in the `strongbox-parent/pom.xml` doesn't do the trick and it caused the following test failures:

```
[ERROR] Tests run: 26, Failures: 3, Errors: 1, Skipped: 0, Time elapsed: 0.212 s <<< FAILURE! - in org.carlspring.strongbox.controllers.users.UserControllerTest
[ERROR] testRetrieveAllUsers(org.carlspring.strongbox.controllers.users.UserControllerTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.IllegalArgumentException: The JSON input text should neither be null nor empty.
	at org.carlspring.strongbox.controllers.users.UserControllerTest.testRetrieveAllUsers(UserControllerTest.java:209)

[ERROR] userNotExistingShouldNotUpdateAccessModel(org.carlspring.strongbox.controllers.users.UserControllerTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <500>.

	at org.carlspring.strongbox.controllers.users.UserControllerTest.getUser(UserControllerTest.java:741)
	at org.carlspring.strongbox.controllers.users.UserControllerTest.userNotExistingShouldNotUpdateAccessModel(UserControllerTest.java:706)

[ERROR] testUpdateAccessModel(org.carlspring.strongbox.controllers.users.UserControllerTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <500>.

	at org.carlspring.strongbox.controllers.users.UserControllerTest.getUser(UserControllerTest.java:741)
	at org.carlspring.strongbox.controllers.users.UserControllerTest.testUpdateAccessModel(UserControllerTest.java:664)

[ERROR] userWithoutViewUserRoleShouldBeAbleToViewHisAccountData(org.carlspring.strongbox.controllers.users.UserControllerTest)  Time elapsed: 0 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <500>.

	at org.carlspring.strongbox.controllers.users.UserControllerTest.userWithoutViewUserRoleShouldBeAbleToViewHisAccountData(UserControllerTest.java:451)
```

These will have to be investigated and sorted out.

**Comment by ptirador** [comment]
I'll pick this up ;)

**Comment by carlspring** [comment]
Great! Thanks! :)

**Comment by ptirador** [comment]
If I upgrade this to version 2.9.2, everything's working ok.
The problem starts from version 2.9.3. If I try to retrieve all users from `UserController`, there's an error about serialization:

`Could not write JSON: Failed to specialize base type java.util.Map<java.lang.String,java.util.Collection<java.lang.String>> as com.orientechnologies.orient.core.db.record.OTrackedMap, problem: Type parameter #1/2 differs; can not specialize java.lang.String with java.lang.Object; nested exception is com.fasterxml.jackson.databind.JsonMappingException: Failed to specialize base type java.util.Map<java.lang.String,java.util.Collection<java.lang.String>> as com.orientechnologies.orient.core.db.record.OTrackedMap, problem: Type parameter #1/2 differs; can not specialize java.lang.String with java.lang.Object (through reference chain: java.util.ArrayList[3]->org.carlspring.strongbox.controllers.users.UserOutput["accessModel"]->org.carlspring.strongbox.users.domain.AccessModel["repositoryPrivileges"])`

The main problem points to an attribute from `BaseUserDto`, which is `AccessModel` class:
```
public class AccessModel
{
    private Map<String, Collection<String>> repositoryPrivileges;

    // maps URL tails including wildcards for regular expressions on set of privileges that was assigned
    // example:
    //      key:    storage0/act-releases-1/pro/redsoft/bar/.*
    //      value:  {ARTIFACTS_VIEW, ARTIFACTS_RESOLVE, ARTIFACTS_DEPLOY}
    private Map<String, Collection<String>> urlToPrivilegesMap;

    private Map<String, Collection<String>> wildCardPrivilegesMap;
[...]
}
```

It seems that there's a serialization problem about these kind of maps...
Any ideas? Is it maybe related to OrientDB?

**Comment by ptirador** [comment]
I've implemented a test which works on 2.9.2, but not from 2.9.3:

` @Test
    public void testSerialization() throws Exception {
        UserOutput user =  new UserOutput();
        AccessModel accessModel = new AccessModel();
        Map<String, Collection<String>> repoPrivilegesMap = **new OTrackedMap(null);**
        String key = "/storages/storage0/releases";
        Collection<String> values = new HashSet<>();
        values.add("ARTIFACTS_RESOLVE");
        values.add("ARTIFACTS_DEPLOY");
        repoPrivilegesMap.put(key, values);
        accessModel.setRepositoryPrivileges(repoPrivilegesMap);
        
        user.setAccessModel(accessModel);

        ObjectMapper mapper = new ObjectMapper();
        System.out.println("Jackson version: " + mapper.version());

        // The following line will fail
        String jsonStr = mapper.writeValueAsString(user);
        
        mapper.readValue(jsonStr, UserOutput.class);
    }`

**Comment by carlspring** [comment]
@fuss86 , @sbespalov ,

Any ideas?

**Comment by carlspring** [comment]
@ptirador ,

Could you maybe check the release notes of the Jackson project and/or their issue tracker and link back, if they have any matching issues...?

**Comment by ptirador** [comment]
@carlspring I'll take a look at the Jackson project and I'll tell you about it.

**Comment by ptirador** [comment]
I've reported an issue in Jackson-databind module: 
[Failed to specialize base type when serializing to JSON](https://github.com/FasterXML/jackson-databind/issues/1964).

**Comment by carlspring** [comment]
@ptirador ,

Thanks for raising this with them! Hopefully, we'll get some help from them soon! The current version of Jackson is vulnerable and we need to upgrade to `2.9.4`, or higher in order to resolve this.

**Comment by carlspring** [comment]
Tried it with `2.9.5` which was released the other day, but that didn't help.

**Comment by carlspring** [comment]
It seems there was a recent fix for this that will be included in `2.9.6`! Would you like to give it a go?

**Comment by ptirador** [comment]
Done!
